llmfoundry/callbacks/async_eval_callback.py

```diff
@@ -0,0 +1,380 @@
+
+    Args:
+        training_run_name: The name of the current training run
+        current_interval: The current interval string of the training run
+
+    Returns:
+        The new run name
+    """
+    name_without_uuid_suffix = training_run_name.rsplit('-', 1)[0]
+
+    max_length = MAX_RUN_NAME_BASE_LENGTH - len(RUN_NAME_PREFIX) - len(
+        current_interval) - 2
+
+    # A run name that is too long will fail a createRun call
+    if len(name_without_uuid_suffix) > max_length:
+        new_name = name_without_uuid_suffix[:max_length]
+        log.warning(
+            f'Training run name {name_without_uuid_suffix} may be too long,' +
+            f' truncating to {new_name}')
+        name_without_uuid_suffix = new_name
+
+    return f'{RUN_NAME_PREFIX}-{current_interval}-{name_without_uuid_suffix}'
+
+
+def get_latest_checkpoint(event: Event, state: State) -> Optional[str]:
+    """Get the latest checkpoint from the training run.
+
+    Args:
+        event: The current run event
+        state: The current state of the training run
+
+    Returns:
+        The path to the latest checkpoint, or None if there is not a latest checkpoint
+    """
+    checkpointer = None
+    for callback in state.callbacks:
+        if isinstance(callback, CheckpointSaver):
+            checkpointer = callback
+            break
+
+    if not checkpointer:
+        log.warning('No checkpoint saver callback found')
+        return None
+
+    if not checkpointer.saved_checkpoints:
+        log.warning('No saved checkpoints found on the checkpointer')
+        return None
+
+    latest = checkpointer.saved_checkpoints[-1]
+    return str(Path(latest).parts[-1])
+
+
+def get_eval_parameters(
+    parameters: Dict[str, Any],
+    checkpoint: str,
+    training_run_name: str,
+    interval: Time,
+) -> Dict[str, Any]:
+    """Get the parameters needed for the eval run.
+
+    Args:
+        parameters: The parameters from the training run
+        checkpoint: The path to the latest checkpoint
+        training_run_name: The name of the training run
+        interval: The current Time interval
+
+    Returns:
+        The parameters needed for the eval run as a dict
+    """
+    looking_for = REQUIRED_PARAMS_FOR_EVAL.copy()
+
+    # Go through all parameters and pull out the ones needed for eval
+    subset_keys = {}
+    for key in parameters:
+        if key in OPTIONAL_PARAMS_FOR_EVAL:
+            subset_keys[key] = parameters[key]
+        elif key in REQUIRED_PARAMS_FOR_EVAL:
+            subset_keys[key] = parameters[key]
+            looking_for.remove(key)
+
+    if looking_for:
+        raise Exception(
+            f'Missing the following required parameters for async eval: {looking_for}'
+        )
+
+    for logger, config in subset_keys.get('loggers', {}).items():
+        if logger == 'wandb':
+            config['group'] = config.pop('name', training_run_name)
+
+            config['init_kwargs'] = config.pop('init_kwargs', {})
```

Comment:
I think it would be a bit nicer to do this through `metadata` arg (currently only present in `train.py` but easy to copy over to `eval.py`). That way it will automatically end up on all the trackers (wandb, mlflow, and run metadata). wdyt?